subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/transform/ArtifactTransformIntegrationTest.groovy

```diff
@@ -163,6 +163,192 @@ class ArtifactTransformIntegrationTest extends AbstractIntegrationSpec {
         file("app/build/libs").listFiles().size() == 1
     }
 
+    def "User gets a reasonable error message when a transformation throws exception"() {
+        given:
+        buildFile << """
+            import org.gradle.api.artifacts.transform.*
+
+            apply plugin: 'java'
+            repositories {
+                mavenCentral()
+            }
+            dependencies {
+                compile 'com.google.guava:guava:19.0'
```

Comment:
Also, we should avoid using the Java plugin in the dependency resolution tests, as this adds another cyclic dependency between dependency resolution and the Java plugin. It happens to work, but we don't want to keep adding more coupling between the 2 projects.